pkg/util/status/status.go

```diff
@@ -0,0 +1,52 @@
+package status
+
+import (
+	"context"
+
+	mdbv1 "github.com/mongodb/mongodb-kubernetes-operator/pkg/apis/mongodb/v1"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type Option interface {
+	ApplyOption(mdb *mdbv1.MongoDB)
```

Comment:
@rodrigovalin this small signature change let's us not need to worry about tracking the state of a single mdb instance, as we can just modify the one that was passed to this function.